transport/src/main/java/io/camunda/zeebe/transport/stream/impl/AggregatedClientStream.java

```diff
@@ -28,23 +31,31 @@
   private final UUID streamId;
   private final LogicalId<M> logicalId;
   private final ClientStreamConsumer streamConsumer;
-  private final Set<MemberId> liveConnections = new HashSet<>();
+  private final ClientStreamMetrics metrics;
 
+  private final Set<MemberId> liveConnections = new HashSet<>();
   private final Int2ObjectHashMap<ClientStream<M>> clientStreams = new Int2ObjectHashMap<>();
 
   private State state;
   private int nextLocalId;
 
   AggregatedClientStream(final UUID streamId, final LogicalId<M> logicalId) {
+    this(streamId, logicalId, ClientStreamMetrics.noop());
+  }
+
+  AggregatedClientStream(
+      final UUID streamId, final LogicalId<M> logicalId, final ClientStreamMetrics metrics) {
     this.streamId = streamId;
     this.logicalId = logicalId;
+    this.metrics = metrics;
 
     streamConsumer = this::push;
     state = State.INITIAL;
   }
 
   void addClient(final ClientStream<M> clientStream) {
     clientStreams.put(clientStream.streamId().localId(), clientStream);
+    metrics.aggregatedStreamCount(clientStreams.size());
```

Comment:
Let's discuss it quickly f2f. Since each instance of `AggregatedClientStream` will record its count as one observation, I was under the impression each observation would be recorded in a bucket. So even if we don't differentiate between streams, we would still have an idea of overall distribution of client count per aggregated stream. e.g.

Streams A, B, C. A records 1, B records 3, and C record 10. This increments the count for each appropriate bucket, even if it's on the same metric. Does it not work this way? :thinking: